cli/azd/cmd/auto_install_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/extensions"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFindFirstNonFlagArg(t *testing.T) {
+	tests := []struct {
+		name     string
+		args     []string
+		expected string
+	}{
+		{
+			name:     "first arg is command",
+			args:     []string{"demo", "--flag", "value"},
+			expected: "demo",
+		},
+		{
+			name:     "command after flags",
+			args:     []string{"--verbose", "demo", "--other"},
+			expected: "demo",
+		},
+		{
+			name:     "only flags",
+			args:     []string{"--help", "--version"},
+			expected: "",
+		},
+		{
+			name:     "empty args",
+			args:     []string{},
+			expected: "",
+		},
+		{
+			name:     "flags with equals",
+			args:     []string{"--config=file", "init", "--template=web"},
+			expected: "init",
+		},
+		{
+			name:     "single character flags",
+			args:     []string{"-v", "-h", "up", "--debug"},
+			expected: "up",
+		},
+		{
+			name:     "command with flag value",
+			args:     []string{"--output", "json", "demo", "subcommand"},
+			expected: "json",
+		},
```

Comment:
Should the expected value be `demo` in this case, since `--output json` belong together? Not sure how hard it would be to handle this, since some flags require additional values to be passed and some don't, like `--help`.